src/_P145_MQxxx.ino

```diff
@@ -0,0 +1,281 @@
+      break;
+    }
+    
+    case PLUGIN_GET_DEVICENAME:
+    {
+      string = F(PLUGIN_NAME_145);
+      break;
+    }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+    {
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_145));
+      break;
+    }
+
+    case PLUGIN_WEBFORM_LOAD:
+    {
+      // FormSelector with all predefined "Sensor - Gas" options
+      String options[P145_MAXTYPES] = {};
+      int optionValues[P145_MAXTYPES] = {};
+      int x = P145_data_struct::getNbrOfTypes();
+      if (x > P145_MAXTYPES) 
+      {
+        x = P145_MAXTYPES;    // Clip to prevent array boundary out of range access
+      }
+      for (int i=0; i<x; i++)
+      {
+        options[i] = concat(P145_data_struct::getTypeName(i), F(" - ")) + (String)P145_data_struct::getGasName(i);
+        optionValues[i] = i; 
+      }
+      addFormSelector(F("Sensor type"), F("plugin_145_sensor"), x, options, optionValues, P145_PCONFIG_SENSORT);
+
+# ifdef ESP32
+      // Analog input selection
+      addRowLabel(F("Analog Pin"));
+      addADC_PinSelect(AdcPinSelectPurpose::ADC_Touch_HallEffect, F("taskdevicepin1"), CONFIG_PIN1);
+# endif // ifdef ESP32
+
+      addFormFloatNumberBox(F("Load Resistance"), F("plugin_145_RLOAD"), P145_PCONFIG_RLOAD, 0.0, 10e6, 2U);
+      addUnit(F("Ohm"));
+      addFormFloatNumberBox(F("R Zero"), F("plugin_145_RZERO"), P145_PCONFIG_RZERO, 0.0, 10e6, 2U);
+      addUnit(F("Ohm"));
+       addFormFloatNumberBox(F("Reference Level"), F("plugin_145_REFLEVEL"), P145_PCONFIG_REF, 0.0, 10e6, 2U);
+      addUnit(F("ppm"));
+      P145_data_struct *P145_data = static_cast<P145_data_struct *>(getPluginTaskData(event->TaskIndex));
+      if (P145_data != nullptr)
+      {
+        float calVal = P145_data->getCalibrationValue();
+        if (calVal > 0.0) 
+        {
+          addFormNote((String)F("Current measurement suggests Rzero= ") + (String)calVal);
+        }
+      }
+
+      addFormSeparator(2);
+      // Auto calibration and Temp/hum compensation flags are bitfields in P145_PCONFIG_FLAGS
+      bool compensate = P145_PCONFIG_FLAGS & 0x0001;
+      bool calibrate = (P145_PCONFIG_FLAGS >> 1) & 0x0001;
+      addFormCheckBox(F("Enable automatic calibration"), F("plugin_145_enable_calibrarion"), calibrate);
+      addFormCheckBox(F("Enable temp/humid compensation"), F("plugin_145_enable_compensation"), compensate);
+      addFormNote(F("If this is enabled, the Temperature and Humidity values below need to be configured."));
+      // temperature
+      addHtml(F("<TR><TD>Temperature:<TD>"));
+      addTaskSelect(F("plugin_145_temperature_task"), P145_PCONFIG_TEMP_TASK);
+      LoadTaskSettings(P145_PCONFIG_TEMP_TASK); // we need to load the values from another task for selection!
+      addHtml(F("<TR><TD>Temperature Value:<TD>"));
+      addTaskValueSelect(F("plugin_145_temperature_value"), P145_PCONFIG_TEMP_VAL, P145_PCONFIG_TEMP_TASK);
+      // humidity
+      addHtml(F("<TR><TD>Humidity:<TD>"));
```

Comment:
See markup functions like `html_TR_TD()` , `html_TR()`, etc.

Reply:
That could even be an `addRowLabel(F("<Label without a colon>"))`